x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -730,6 +760,38 @@ export class TaskRunner<
     return { executionStatus, executionMetrics };
   }
 
+  async beforeRun(): Promise<BeforeRunResult<RawRule>> {
```

Comment:
Seems like it would be better to call this `getRuleData()` or similiar.

Reply:
As it is called by the TaskManager for all the task types (rule, action, etc.), has to be a generic name.
LoadData maybe?

Reply:
done